src/app/needs/next-actions/components/NextActionsSection.tsx

```diff
@@ -1,38 +1,102 @@
 "use client";
 
-import React from "react";
+import React, { useState } from "react";
 import Button from "@/ui/shared/Button";
+import Modal from "@/ui/shared/Modal";
 import { NeedDocument } from "./NextActionsDisplay";
 import { NextActionDocument } from "./NextActionsDisplay";
+import { PlusCircleIcon } from "@heroicons/react/24/outline";
 
-interface NextActionsSectionProps {
+interface SectionProps {
   need: NeedDocument;
   actions: NextActionDocument[];
   onToggleAction: (action: NextActionDocument) => Promise<void>;
+  handleAddAction: (newAction: string, need: NeedDocument) => Promise<void>;
 }
 
 export default function NextActionsSection({
+  need,
   actions,
   onToggleAction,
-}: NextActionsSectionProps) {
-  return (
+  handleAddAction,
+}: SectionProps) {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [newAction, setNewAction] = useState<string>("");
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewAction(e.target.value)
+  };
+
+  const handleSubmitAction = async () => {
+    await handleAddAction(newAction, need);
+    setModalOpen(false);
+  }
+
+  /* const handleAddAction = async () => {
+    if (newAction.trim()) {
+      const newActionDocument = {
+        id: crypto.randomUUID(),
+        name: newAction.trim(),
+        need: need.id,
+        selectedTimestamps: [],
+        selectedExpiry: new Date().toISOString(),
+        timestamp: new Date().toISOString(),
+      }
+
+      try {
+        await database.addToDb("next_actions", newActionDocument);
+
+        console.groupCollapsed("Action Created");
+        console.log(`Name: ${newActionDocument.name}`);
+        console.log(`Need: ${newActionDocument.need}`);
+        console.groupEnd();
+
+        setTriggerRerender(triggerRerender + 1);
+      } catch (error) {
+        console.error("Error creating Action:", error);
+      }
+      
+      setModalOpen(false);
+    }
+  }; */
+
```

Comment:
```suggestion
```
If we don't need, delete [will not block merge if it is needed for a future use]

Reply:
Good catch